test/parse-text-files.ts

```diff
@@ -22,126 +22,109 @@ before(() => {
   setup();
 });
 
-// tslint:disable:no-unused-expression
-// .true triggers ts-lint failure, but is valid chai
-
 describe('Parse text files function', () => {
-  it('Parses map objects into Change object', () => {
+  it('should parse map objects into Change object', () => {
     const userFilesMap = new Map<string, string | null>();
     userFilesMap.set('src/index.js', "console.log('hello index!'");
     userFilesMap.set('src/foo.js', "console.log('hello foo!'");
     userFilesMap.set('src/deleted.js', null);
 
     const parsedMap = new Map<string, FileData>();
-    parsedMap.set('src/index.js', {
-      content: "console.log('hello index!'",
-      mode: '100644',
-    });
-    parsedMap.set('src/foo.js', {
-      content: "console.log('hello foo!'",
-      mode: '100644',
-    });
-    parsedMap.set('src/deleted.js', {
-      content: null,
-      mode: '100644',
-    });
+    parsedMap.set(
+      'src/index.js',
+      new FileData("console.log('hello index!'", '100644')
+    );
+    parsedMap.set(
+      'src/foo.js',
+      new FileData("console.log('hello foo!'", '100644')
+    );
+    parsedMap.set('src/deleted.js', new FileData(null, '100644'));
     // tests
     const changes = parseTextFiles(userFilesMap);
-    expect(changes).to.deep.equal(parsedMap);
```

Comment:
I now fully distrust the deep equal implementation in Chai.  The returned object was actually a set with class instances, not naked objects.